tsconfig.json

```diff
@@ -0,0 +1,109 @@
+{
```

Comment:
I think it can be useful to extend a widely-used standard configuration rather than writing one out explicitly. So maybe you can use https://github.com/tsconfig/bases as we do in recent projects, using for example
```
{
  "extends": ["@tsconfig/strictest/tsconfig", "@tsconfig/node18/tsconfig"]
}
```

@gabro What do you think? I added you as reviewer too, especially for what concerns the first setup steps and config choices.